cli/flox/src/commands/show.rs

```diff
@@ -37,6 +38,10 @@ impl Show {
     pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("show");
 
+        if self.all {
+            message::warning("'--all' is now the default and the flag has been deprecated.");
```

Comment:
I think we should leave the flag as hidden so that existing (particularly scripted) uses don't cause an error but open to suggestion about whether the warning is necessary. cc @ghudgins @jennymahmoudi

Reply:
I think the warning on `--all` is a good idea. At some point, it will be actually deprecated, right?

Reply:
Yeah, it would be removed later.

Is there prior art for doing that? Leave a TODO, file a new issue, or something else?

Reply:
> Leave a TODO, file a new issue, or something else?

yes

I guess for higher visible things an issue is good so if you get the warning you can find and subscribe to the respective issue. (personally i'd like the warning to include a link to that issue as well).
An issue is also usually a better reminder for us and slightly less likely to be forgotten about then a TODO comment.

Then again we have not yet converged to any single action in this case.